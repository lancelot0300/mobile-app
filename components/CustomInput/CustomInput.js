import React from 'react'
import {View, TextInput, StyleSheet, Image } from 'react-native'

const CustomInput = ({value, placeholder, onChangeText, style, img , secureTextEntry}) => {


  return (
    <View>
        <View style={styles.imgCont}>
            <Image style={styles.img} source={img}/>
        </View>
        <TextInput 
             value={value}
             onChangeText={onChangeText}
             placeholder={placeholder}
             style={{...styles.input, ...style}}
            secureTextEntry={secureTextEntry}
        />
    </View>
  )
}

const styles = StyleSheet.create({
    input: {
        height: 40,
        margin: 12,
        borderWidth: 1,
        padding: 10,
        borderRadius: 10,
        borderColor: 'white',
        color: 'white',
        minWidth: 200,
        paddingLeft: 45,
    },
    imgCont: {
        position: 'absolute',
        width: 30,
        left: 20,
        top: 20,
        zIndex: 1,
        borderRightWidth: 1,
        borderColor: 'white',
    },
    img: {
        position: 'relative',
        width: 25,
        height: 25,
        padding: 10,
        paddingRight: 10,
    }

});

export default CustomInput