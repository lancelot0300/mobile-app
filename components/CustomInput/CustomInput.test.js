import React from 'react';
import renderer from 'react-test-renderer';

import CustomInput from './CustomInput';



describe('<CustomInput />', () => {

    it('renders correctly', () => {
        const tree = renderer.create(<CustomInput 
            value={'Text'}
            onChangeText={() => {}}
            placeholder={'Placeholder'}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    }
    );


});