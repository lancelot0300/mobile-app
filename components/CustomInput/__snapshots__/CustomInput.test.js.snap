// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomInput /> renders correctly 1`] = `
<View>
  <View
    style={
      {
        "borderColor": "white",
        "borderRightWidth": 1,
        "left": 20,
        "position": "absolute",
        "top": 20,
        "width": 30,
        "zIndex": 1,
      }
    }
  >
    <Image
      style={
        {
          "height": 25,
          "padding": 10,
          "paddingRight": 10,
          "position": "relative",
          "width": 25,
        }
      }
    />
  </View>
  <TextInput
    onChangeText={[Function]}
    placeholder="Placeholder"
    style={
      {
        "borderColor": "white",
        "borderRadius": 10,
        "borderWidth": 1,
        "color": "white",
        "height": 40,
        "margin": 12,
        "minWidth": 200,
        "padding": 10,
        "paddingLeft": 45,
      }
    }
    value="Text"
  />
</View>
`;
